### PR TITLE
Chore/create integration test

### PR DESCRIPTION
**Pull request by andrewtran1995**
Created rudimentary integration test that calls shell script files used in normal execution, checking the curl command's output concerning the server's response.

Also, work from the chore/document_aruco_alg somehow creeped in here, so ... *shrug*. Not quite sure how to revert it, but I don't believe there'd be a merge conflict when that branch is merged later on with this one.

### CODE REVIEW

File: systemTests/testImageUpload.py [hunk 1]
```diff
@@ -1,7 +1,43 @@
...
+        flask_server = subprocess.Popen(ScriptWrapper.runFlaskServerPath())
+        # wait for Flask Server and controller to connect
+        time.sleep(5)
+
+        # initiate image upload via curl command
+        print("~~~ Calling curl command ~~~")
+        try:
+            output = subprocess.check_output(ScriptWrapper.curlCommandPath(), timeout=25)
+        except TimeoutExpired:
+            self.fail("Image upload timed out!")
```
Comment (joyoyoyoyoyo) [inline systemTests/testImageUpload.py:31]: I think there might be a memory leak here.
Do we need to call `controller.kill()` or `flask_server.kill()`, when these Exceptions or assertions are called or terminate the program and test? 

On an aside: For non-integration tests, `controller.kill()` and `flask_server.kill()` can go in a tear down method that the potentially offered by the Python unit test framework.

File: ImP/imageProcessing/README.md [hunk 2]
```diff
@@ -38,3 +38,51 @@ Parameters:
...
+    // copy to output arrays
+    _copyVector2Output(candidates, _corners);
+    Mat(ids).copyTo(_ids);
+
+    /// STEP 4: Corner refinement
+    if(_params->doCornerRefinement) {
+        CV_Assert(_params->cornerRefinementWinSize > 0 && _params->cornerRefinementMaxIterations > 0 &&
+                  _params->cornerRefinementMinAccuracy > 0);
+
+        //// do corner refinement for each of the detected markers
```
Comment (joyoyoyoyoyo) [inline ImP/imageProcessing/README.md]: On future code, can we add the reason behind certain code being commented out?

File: ImP/imageProcessing/arucoMarkerDetectionAlgorithm.md [hunk 3]
```diff
@@ -61,7 +51,144 @@ static void _detectCandidates(InputArray _image, vector< vector< Point2f > >& ca
...
+
+```
+   // number of window sizes (scales) to apply adaptive thresholding
+   int nScales =  (params->adaptiveThreshWinSizeMax - params->adaptiveThreshWinSizeMin) /
+                     params->adaptiveThreshWinSizeStep + 1;
+
+   vector< vector< vector< Point2f > > > candidatesArrays((size_t) nScales);
+   vector< vector< vector< Point > > > contoursArrays((size_t) nScales);
+```
+[Threshold](https://en.wikipedia.org/wiki/Thresholding_(image_processing)) the image at different scales to find different markers.
```
Comment (joyoyoyoyoyo) [inline ImP/imageProcessing/arucoMarkerDetectionAlgorithm.md:82]: Can we comment in the algorithms being used or link the Math(if we write it ourselves)?

**Comment by joyoyoyoyoyo** [comment]
Really amazing stuff. I liked it a lot :smile: :+1:

**Comment by andrewtran1995** [comment]
Cool -- I'll implement the improvements you suggested in another PR then.

File: ImP/imageProcessing/arucoMarkerDetectionAlgorithm.md (hunk 3 above)
Comment (andrewtran1995) [reply ImP/imageProcessing/arucoMarkerDetectionAlgorithm.md:82]: I think that should be alright, but let me know ahead of time if you do so!

**Pull request merged**